mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -106,6 +104,46 @@ def on_epoch_end(self, trainer, pl_module):
                     if isinstance(callback, pl.callbacks.early_stopping.EarlyStopping):
                         self._early_stop_check(callback)
 
+            def on_train_epoch_end(self, trainer, pl_module, _):
```

Comment:
Rather than checking `_is_pytorch_lightning_older_than_1_2_0` each time one of these callback methods is invoked,  we could conditionally define these callback methods: e.g.,

```
if LooseVersion(pl.__version) >= LooseVersion("1.2.0"):

    def on_train_epoch_end(...)
    def on_validation_epoch_end(...)

else:

    def on_epoch_end(...)
``` 

This may be a bit easier to read / reason about and will remove some boolean checks when executing callback hooks.

Reply:
oh I didn't know we could do that. Let me fix that

Reply:
@dbczumar Fixed!